xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -1639,41 +1643,22 @@ static CdsUpdate processCluster(Cluster cluster, Set<String> retainedEdsResource
     }
     CdsUpdate.Builder updateBuilder = structOrError.getStruct();
 
-    if (cluster.getLbPolicy() == LbPolicy.RING_HASH) {
-      RingHashLbConfig lbConfig = cluster.getRingHashLbConfig();
-      long minRingSize =
-          lbConfig.hasMinimumRingSize()
-              ? lbConfig.getMinimumRingSize().getValue()
-              : DEFAULT_RING_HASH_LB_POLICY_MIN_RING_SIZE;
-      long maxRingSize =
-          lbConfig.hasMaximumRingSize()
-              ? lbConfig.getMaximumRingSize().getValue()
-              : DEFAULT_RING_HASH_LB_POLICY_MAX_RING_SIZE;
-      if (lbConfig.getHashFunction() != RingHashLbConfig.HashFunction.XX_HASH
-          || minRingSize > maxRingSize
-          || maxRingSize > MAX_RING_HASH_LB_POLICY_RING_SIZE) {
-        throw new ResourceInvalidException(
-            "Cluster " + cluster.getName() + ": invalid ring_hash_lb_config: " + lbConfig);
-      }
-      updateBuilder.ringHashLbPolicy(minRingSize, maxRingSize);
-    } else if (cluster.getLbPolicy() == LbPolicy.ROUND_ROBIN) {
-      updateBuilder.roundRobinLbPolicy();
-    } else if (enableLeastRequest && cluster.getLbPolicy() == LbPolicy.LEAST_REQUEST) {
-      LeastRequestLbConfig lbConfig =  cluster.getLeastRequestLbConfig();
-      int choiceCount =
-              lbConfig.hasChoiceCount()
-                ? lbConfig.getChoiceCount().getValue()
-                : DEFAULT_LEAST_REQUEST_CHOICE_COUNT;
-      if (choiceCount < DEFAULT_LEAST_REQUEST_CHOICE_COUNT) {
-        throw new ResourceInvalidException(
-                "Cluster " + cluster.getName() + ": invalid least_request_lb_config: " + lbConfig);
-      }
-      updateBuilder.leastRequestLbPolicy(choiceCount);
-    } else {
-      throw new ResourceInvalidException(
-          "Cluster " + cluster.getName() + ": unsupported lb policy: " + cluster.getLbPolicy());
+    // TODO: If load_balancing_policy is set in Cluster use it for LB config, otherwise fall back
+    // to using the legacy lb_policy field.
+    ImmutableMap<String, ?> lbPolicyConfig = LegacyLoadBalancerConfigBuilder.forCluster(cluster,
+        enableLeastRequest).build();
+
+    // Validate the LB config by trying to parse it with the corresponding LB provider.
+    LbConfig lbConfig = ServiceConfigUtil.unwrapLoadBalancingConfig(lbPolicyConfig);
+    NameResolver.ConfigOrError configOrError = loadBalancerRegistry.getProvider(
+        lbConfig.getPolicyName()).parseLoadBalancingPolicyConfig(
```

Comment:
nit: style guide prefers wrapping at the higher syntactic levels. Instead of wrapping in the middle of arguments and having two different method calls' arguments aligned, it seems this would read better by wrapping earlier:
```java
    NameResolver.ConfigOrError configOrError = loadBalancerRegistry
        .getProvider(lbConfig.getPolicyName())
        .parseLoadBalancingPolicyConfig(lbConfig.getRawConfigValue());
```